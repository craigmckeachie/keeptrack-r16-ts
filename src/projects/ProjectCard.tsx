import { Project } from './Project';
import React from 'react';
import { Link } from 'react-router-dom';

interface ProjectCardProps {
  project: Project;
  onEdit: (project: Project) => void;
}

function formatDescription(description: string): string {
  return description.substring(0, 60) + '...';
}

function ProjectCard(props: ProjectCardProps) {
  const { project, onEdit } = props;
  const handleEditClick = (projectBeingEdited: Project) => {
    onEdit(projectBeingEdited);
  };
  return (
    <div className="card">
      <img src={project.imageUrl} alt={project.name} />
      <section className="section dark">
        <Link to={'/projects/' + project.id}>
          <h5 className="strong">
            <strong>{project.name}</strong>
          </h5>
          <p>{formatDescription(project.description)}</p>
        </Link>
        <p>Budget : {project.budget.toLocaleString()}</p>
        <button
          className="bordered"
          onClick={() => {
            handleEditClick(project);
          }}
        >
          <span className="icon-edit " />
          Edit
        </button>
      </section>
    </div>
  );
}

export default ProjectCard;
