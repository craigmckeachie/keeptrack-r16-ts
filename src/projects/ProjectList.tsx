import React from 'react';
import { Project } from './Project';

interface ProjectListProps {
  projects: Project[];
}

class ProjectList extends React.Component<ProjectListProps> {
  render() {
    const { projects } = this.props;
    return <pre>{JSON.stringify(projects, null, ' ')}</pre>;
  }
}

export default ProjectList;
