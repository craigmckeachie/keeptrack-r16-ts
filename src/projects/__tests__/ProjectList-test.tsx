import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import ProjectList, { ProjectListProps } from '../ProjectList';
import { Project } from '../Project';
import { MOCK_PROJECTS } from '../MockProjects';
import { ProjectCardProps } from '../ProjectCard';

describe('<ProjectList />', () => {
  let wrapper: ShallowWrapper;
  let mockProjects: Project[];

  beforeEach(() => {
    mockProjects = MOCK_PROJECTS;
    wrapper = shallow(<ProjectList projects={mockProjects} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('renders <ProjectCard/>s', () => {
    const projectCardWrapper = wrapper.find('ProjectCard');
    expect(projectCardWrapper.length).toBe(mockProjects.length);
  });

  test('render <ProjectForm> for editingProject', () => {
    wrapper.setState({ editingProject: mockProjects[2] });
    // expect(wrapper.instance().state.editingProject).toBe(mockProjects[2]);
    expect(wrapper.find('Connect(ProjectForm)').length).toBe(1);
  });

  test('snapshot', () => {
    // wrapper.setState({ editingProject: mockProjects[2] });
    expect(wrapper).toMatchSnapshot();
  });
});
