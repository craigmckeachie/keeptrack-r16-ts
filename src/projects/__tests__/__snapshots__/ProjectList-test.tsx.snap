// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectList /> snapshot 1`] = `ShallowWrapper {}`;
